sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -78,9 +78,42 @@ case class ProjectExec(projectList: Seq[NamedExpression], child: SparkPlan)
     }
   }
 
-  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
```

Comment:
Well, if we wanna support this, we should support all the case. e.g.
```
Project [(a#1 + b#2) as c#3, b#2]
```
and then the parent operator requires partitioning on `c#3 - b#2`. Since `(c#3 - b#2).samanticEquals(a#1)` is false, additional ShuffleExchange will be added.

This is not trivial and we should have a holistic solution.